math/integral_approximation.cpp

```diff
@@ -0,0 +1,111 @@
+/**
+ * @file
+ * @brief Algorithm to compute integral approximation of function

```

Comment:
Provide a Wikipedia link in Markdown format (if available/possible).
If there's no Wikipedia link, add another web source for algorithm explanation.

Also, add a detailed/long description of the algorithm (use `@details` for this).